socket_linux.go

```diff
@@ -26,13 +31,26 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 		return nil, err
 	}
 
-	sockaddr := unix.SockaddrInet4{
-		Port: addr.Port,
-	}
-	if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
-		panic("did not copy enough bytes of ip address")
+	var sa unix.Sockaddr
+	if domain == unix.AF_INET {
+		sockaddr := &unix.SockaddrInet4{
+			Port: addr.Port,
+		}
+		if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
```

Comment:
Ugh. As I was reading the various functions around this code (in golang) I was confused why there were different representations of the same thing. Oh well!

Thanks for tolerating my suggestions. :)